pro/src/new_components/ProfileForm/ProfileForm.tsx

```diff
@@ -10,33 +10,42 @@ import styles from './ProfileForm.module.scss'
 
 export interface IProfileFormProps {
   title: string
-  subtitle: string
+  subtitleFormat: (values: any) => string
   fields: Array<JSX.Element>
   validationSchema: AnySchema
   banner?: JSX.Element
-  shouldDisplayBanner: boolean
-  onFormSubmit: (values: any) => void
+  shouldDisplayBanner?: boolean
+  adapter: (values: any) => Promise<any>
   initialValues: { [id: string]: string }
 }
 
 const ProfileForm = ({
   title,
-  subtitle,
+  subtitleFormat,
   fields,
   validationSchema,
   initialValues,
   shouldDisplayBanner = false,
   banner,
-  onFormSubmit,
+  adapter,
 }: IProfileFormProps): JSX.Element => {
   const [isFormVisible, setIsFormVisible] = useState(false)
+  const [subTitle, setSubTitle] = useState(subtitleFormat(initialValues))
   const toggleFormVisible = () => {
     setIsFormVisible(oldValue => !oldValue)
   }
   const onSubmit = (values: any) => {
-    onFormSubmit(values)
+    adapter(values)
+      .then(response => {
+        toggleFormVisible()
+        formik.setValues(response.payload)
+        setSubTitle(subtitleFormat(response.payload))
+      })
+      .catch(errors => {
+        for (const field in errors)
+          formik.setFieldError(field, (errors as any)[field])
+      })
```

Comment:
L'adapter ne throw jamais d'erreurs.
Il renvoit toujours un json contenant `isOK: true` ou bien `isOk: false`

```suggestion
    const response = updateUser(values)
    if (response.isOk) {
        // handle ok api response
        // by using response.payload
    } else {
        // handle failing api response
        // by using response.message for the error message
        // or response.payload.errors if api return errors (have to be define in the adapter)
    }
```